Odata-docs/webapi-8/routing-overview.md

```diff
@@ -0,0 +1,365 @@
+            "kind": "EntitySet",
+            "url": "Customers"
+        }
+    ]
+}
+```
+
+Resources exposed by an OData service are addressable by the corresponding **resource path**. Examples of resources might customers, a single customer, and so forth. Examples of addressable aspects of these resources as exposed by the data model might be: collections of entities, a single entity, properties, links, operations, and so on. As an illustration, to address a collection of customer entities via the entity set, the relevant resource path would be `Customers`.
+#### Request
+```http
+GET http://localhost:5000/odata/Customers
+```
+#### Response
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1"
+        },
+        {
+            "Id": 2,
+            "Name": "Customer 2"
+        }
+    ]
+}
+```
+
+The **query options** part of an OData URL specifies three types of information: system query options, customer query options and parameter aliases.
+
+System query options are query string parameters that control the amount and order of the data returned for the resource identified by the URL. The names of all system query options are prefixed with a dollar (`$`) symbol by default. However, the OData standard does not make it mandatory to prefix system query options with a dollar symbol so ASP.NET Core OData 8 allows you to relax or enforce the requirement for the prefix. Examples of system query options are: `$filter`, `$orderby`, and `$top`.
+
+Custom query options provide an extensible mechanism for service-specific information to be placed in a URL query string. Customer query options MUST NOT begin with a `$` or `@` character.
+
+Parameter aliases can be used in place of literal values in entity keys, function parameters, or within a `$filter` or `$orderby` expression. Parameter aliases must start with an `@` character. Parameter alias usage example: `GetOrdersInRange(dateStart=@dateStart,dateEnd=@dateEnd)?@dateStart=2022-11-07&@dateEnd=2022-12-07`
+
+```http
+GET http://localhost:5000/odata/Customers?$select=Name
+```
+#### Response
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Customers(Name)",
+    "value": [
+        {
+            "Name": "Customer 1"
+        },
+        {
+            "Name": "Customer 2"
+        }
+    ]
+}
+```
+
+## OData routing concepts
+To opt in to OData routing, a controller or controller action needs to be decorated with `ODataAttributeRouting` attribute as illustrated below:
+
+#### `ODataAttributeRouting` on controller
+```csharp
+[ODataAttributeRouting]
+public class CustomersController : ControllerBase
+{
+    [EnableQuery]
+    public ActionResult Get()
+    {
+        // ...
+    }
+}
+```
+
+#### `ODataAttributeRouting` on controller action
+```csharp
+public class CustomersController : ControllerBase
+{
+    [ODataAttributeRouting]
```

Comment:
I thought in 8.x we can use aspnetcore's Route attribute and we don't need to use the ODataAttributeRouting attribute.